src/Adapter/SqlManager/SqlQueryValidator.php

```diff
@@ -0,0 +1,287 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\SqlManager;
+
+use ErrorException;
+use PrestaShop\PrestaShop\Adapter\Entity\RequestSql;
+
+/**
+ * Class SqlQueryValidator is responsible for validating Request SQL model data
+ */
+class SqlQueryValidator
+{
+    /**
+     * Check if SQL is valid for Reqest SQL model.
+     * Only "Select" sqls should be valid.
+     *
+     * @param string $sql
+     *
+     * @return array Array of errors if any
+     */
+    public function validate($sql)
+    {
+        $errors = [];
+
+        try {
+            $requestSql = new RequestSql();
+            $parser = $requestSql->parsingSql($sql);
+            $validate = $requestSql->validateParser($parser, false, $sql);
+
+            if (!$validate || count($requestSql->error_sql)) {
+                $errors = $this->getErrors($requestSql->error_sql);
+            }
+        } catch (ErrorException $e) {
+            $errors[] = [
+                'key' => 'Bad SQL query',
+                'parameters' => [],
+                'domain' => 'Admin.Notifications.Error',
+            ];
+        }
+
+        return $errors;
+    }
+
+    /**
+     * Get request sql errors
+     *
+     * @param array $sqlErrors
+     *
+     * @return array
+     */
+    private function getErrors(array $sqlErrors)
+    {
+        $errors = [];
+
+        foreach ($sqlErrors as $key => $sqlError) {
+            switch ($key) {
+                case 'checkedFrom':
+                    if (isset($sqlError['table'])) {
+                        $errors[] = [
+                            'key' => 'The "%tablename%" table does not exist.',
+                            'parameters' => [
+                                '%tablename%' => $sqlError['table'],
+                            ],
+                            'domain' => 'Admin.Advparameters.Notification',
+                        ];
+                    } elseif (isset($sqlError['attribut'])) {
```

Comment:
`attribut` ? should be `attribute` no?

Reply:
actually it is `attribut`, it comes deep from legacy.